clients/src/main/java/org/apache/kafka/common/metrics/Sensor.java

```diff
@@ -300,6 +300,18 @@ public boolean hasExpired() {
         return unmodifiableList(new LinkedList<>(this.metrics.values()));
     }
 
+    /**
+     * Find a metric attached to this sensor by name.
+     */
+    public synchronized KafkaMetric findByName(String name) {
```

Comment:
I believe this is a public class. Since the method is only used in a test, perhaps we could look up the metric in the test?

Reply:
Fair enough. I'm almost tempted to do a KIP for this since accessing the metrics is painful at the moment. Anyway, I'll just add a test lookup method in the test as suggested.